localstack/services/apigateway/integration.py

```diff
@@ -394,6 +395,133 @@ def invoke(self, invocation_context: ApiInvocationContext):
         return invocation_context.response
 
 
+class KinesisIntegration(BackendIntegration):
+    def invoke(self, invocation_context: ApiInvocationContext):
+        integration = invocation_context.integration
+        integration_type_orig = integration.get("type") or integration.get("integrationType") or ""
+        integration_type = integration_type_orig.upper()
+        uri = integration.get("uri") or integration.get("integrationUri") or ""
+
+        if uri.endswith("kinesis:action/PutRecord"):
+            target = "Kinesis_20131202.PutRecord"
+        elif uri.endswith("kinesis:action/PutRecords"):
+            target = "Kinesis_20131202.PutRecords"
+        elif uri.endswith("kinesis:action/ListStreams"):
+            target = "Kinesis_20131202.ListStreams"
+        else:
+            LOG.info(
+                f"Unexpected API Gateway integration URI '{uri}' for integration type {integration_type}",
+            )
+            target = ""
+
+        try:
+            invocation_context.context = helpers.get_event_request_context(invocation_context)
+            invocation_context.stage_variables = helpers.get_stage_variables(invocation_context)
+            request_templates = RequestTemplates()
+            payload = request_templates.render(invocation_context)
+
+        except Exception as e:
+            LOG.warning("Unable to convert API Gateway payload to str", e)
+            raise
+
+        # forward records to target kinesis stream
+        headers = aws_stack.mock_aws_request_headers(
+            service="kinesis", region_name=invocation_context.region_name
+        )
+        headers["X-Amz-Target"] = target
+
+        result = common.make_http_request(
+            url=config.service_url("kinesis"), data=payload, headers=headers, method="POST"
+        )
+
+        # apply response template
+        invocation_context.response = result
+        response_templates = ResponseTemplates()
```

Comment:
response template are already in the base class. The only thing required here is call `response_templates.render(invocation_context)`

Reply:
Good catch, thanks for the pointer!